lib/from_openstudio/schedule/ruleset.rb

```diff
@@ -0,0 +1,114 @@
+# Energy, LLC, Ladybug Tools LLC and other contributors. All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# (1) Redistributions of source code must retain the above copyright notice,
+# this list of conditions and the following disclaimer.
+#
+# (2) Redistributions in binary form must reproduce the above copyright notice,
+# this list of conditions and the following disclaimer in the documentation
+# and/or other materials provided with the distribution.
+#
+# (3) Neither the name of the copyright holder nor the names of any contributors
+# may be used to endorse or promote products derived from this software without
+# specific prior written permission from the respective party.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDER(S) AND ANY CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+# THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER(S), ANY CONTRIBUTORS, THE
+# UNITED STATES GOVERNMENT, OR THE UNITED STATES DEPARTMENT OF ENERGY, NOR ANY OF
+# THEIR EMPLOYEES, BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT
+# OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+# STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+# OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+# *******************************************************************************
+
+require 'honeybee/schedule/ruleset'
+require 'from_openstudio/model_object'
+
+module Honeybee
+  class ScheduleRulesetAbridged < ModelObject
+
+    def self.from_schedule_ruleset(schedule_ruleset)
+      # create an empty hash
+      hash = {}
+      hash[:type] = 'ScheduleRulesetAbridged'
+      # set hash values from OpenStudio Object
+      hash[:identifier] = schedule_ruleset.nameString
+      # check if boost optional object is empty
+      hash[:default_day_schedule] = schedule_ruleset.defaultDaySchedule.nameString 
+      hash[:summer_designday_schedule] = schedule_ruleset.summerDesignDaySchedule.nameString
+      hash[:winter_designday_schedule] = schedule_ruleset.winterDesignDaySchedule.nameString
+      hash[:holiday_schedule] = schedule_ruleset.holidaySchedule.nameString
+
+      schedule_days = [schedule_ruleset.defaultDaySchedule, schedule_ruleset.summerDesignDaySchedule, schedule_ruleset.winterDesignDaySchedule, schedule_ruleset.holidaySchedule]
+      hash[:day_schedules] = []
+      schedule_days.each do |schedule_day|
+        hash[:day_schedules] << ScheduleRulesetAbridged.from_day_schedule(schedule_day)
+      end
```

Comment:
We also need to add all of the ScheduleDays that are used by each rule. And we should probably check for the case that the same ScheduleDay is used multiple times. I think this only needs to be a check for the object nameString and we can keep a set of day schedule names to check against.